src/publisher/aws_events.py

```diff
@@ -1,52 +1,108 @@
+from ordered_set import OrderedSet
 import json
 from django.conf import settings
 import boto3
 from publisher import relation_logic as relationships
-from publisher.utils import lmap
+from functools import wraps
+
+from multiprocessing import SimpleQueue
 
 import logging
 
 LOG = logging.getLogger(__name__)
 
-def sns_topic_arn(**overrides):
+def sns_topic_arn():
     "returns an arn path to an AWS event bus. this is used to connect and send/receive events"
     vals = {}
     vals.update(settings.EVENT_BUS)
-    vals.update(overrides)
     # ll: arn:aws:sns:us-east-1:112634557572:bus-articles--ci
     arn = "arn:aws:sns:{region}:{subscriber}:{name}--{env}".format(**vals)
     LOG.info("using topic arn: %s", arn)
     return arn
 
-def event_bus_conn(**overrides):
+def event_bus_conn():
     sns = boto3.resource('sns')
-    return sns.Topic(sns_topic_arn(**overrides))
+    return sns.Topic(sns_topic_arn())
+
+#
+#
+#
+
+SAFEWORD = START = STOP = 'cacao'
+
+def defer(fn):
+    """calls function normally until safeword is received then buffers all requests until the safeword is called again.
```

Comment:
This looks like it needs three primitives to call from outside:

- open()
- do()
- close()

rather than a single function; the next person passing from here will thank you